aten/src/ATen/core/PhiloxRNGEngine.h

```diff
@@ -105,15 +105,14 @@ class philox_engine {
     #ifdef __CUDA_ARCH__
     AT_ASSERT(false, "Unsupported invocation of randn on CUDA");
     #endif
-    reset_state(); // Reset state for randn - a little wasteful, but easier to ensure correctness.
     detail::UINT4 counter = counter_;
     detail::UINT2 key = key_;
     detail::UINT4 i = rand(counter, key, n_rounds);
```

Comment:
it looks like `rand` doesn't call `incr()` the way operator `()` does, why is that? We are supposed to advance counter for each call.

Reply:
Done, perhaps @voznesenskym could help provide some context why `reset_state()` instead of `incr()` for `randn` were there.

Reply:
Hi @voznesenskym, @ngimel ~ bump on this. Is this good to go?